src/Cache/Mutator/AddHeaderResponseMutator.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Http\Client\Common\Plugin\Cache\Mutator;
+
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+
+/**
+ * Adds a header if the response came from cache.
+ *
+ * @author Iain Connor <iain.connor@priceline.com>
+ */
+class AddHeaderResponseMutator implements ResponseMutator
+{
+    /** @var string */
+    private $headerName;
+
+    /**
+     * @param string $headerName
+     */
+    public function __construct($headerName = 'X-From-Php-Http-Cache')
+    {
+        $this->headerName = $headerName;
+    }
+
+    /**
+     * Mutate the response depending on the cache status.
+     *
+     * @param ResponseInterface $response
+     * @param bool              $cacheHit
+     *
+     * @return string
+     */
+    public function mutate(RequestInterface $request, ResponseInterface $response, $cacheHit)
```

Comment:
is there any more information that we could pass in instead of the boolean? like the $cacheItem or such? then this could be used to extract additional debugging information if wanted.

Reply:
Good suggestion, updated.

Reply:
For the record, I'm going to keep the `cacheHit` parameter as well, so the consumer doesn't have to reverse-engineer the logic regarding expiry. Checking `cacheItem->isHit()` won't be sufficient.